pkg/router/template/router.go

```diff
@@ -841,23 +844,26 @@ func generateDestCertKey(config *ServiceAliasConfig) string {
 // getServiceUnits returns a map of service keys to their weights.
 // Weight suggests the % of traffic that a given service will receive
 // compared to other services pointed to by the route.
-func getServiceUnits(route *routeapi.Route) map[string]int32 {
+func getServiceUnits(route *routeapi.Route) (map[string]int32, bool) {
 	serviceUnits := make(map[string]int32)
+	hasWeights := false
 	key := fmt.Sprintf("%s/%s", route.Namespace, route.Spec.To.Name)
 	if route.Spec.To.Weight == nil {
 		serviceUnits[key] = 0
 	} else {
 		serviceUnits[key] = *route.Spec.To.Weight
+		hasWeights = true
```

Comment:
Shouldn't this check be that there is more than one service?  A route with weight 100 and one backend should behave exactly the same as a route with weight 1 and one backend.